tests/test_parallels.py

```diff
@@ -131,3 +132,139 @@ def test_parallel_success_on_one():
+
+    different_policy = py_trees.common.ParallelPolicy.SuccessOnSelected(
+            children=[running1, running2]
+    )
+    empty_policy = py_trees.common.ParallelPolicy.SuccessOnSelected(
+            children=[]
+    )
+    parallel = py_trees.composites.Parallel(
+        name="Parallel",
+        children=[running1, running3]
+    )
+    for policy in [different_policy, empty_policy]:
+        print("")
+        print(console.cyan + "Policy Children: " + console.yellow + str([c.name for c in policy.children]) + console.reset)
+        print(console.cyan + "Parallel Children: " + console.yellow + str([c.name for c in parallel.children]) + console.reset)
+        parallel.policy = policy
+        print("\n--------- Assertions ---------\n")
+        print("setup() returns 'False' due to invalid configuration")
+        result = parallel.setup(timeout=5.0)
+        assert(not result)
+        print("initialise() raises a 'RuntimeError' due to invalid configuration")
+        with nose.tools.assert_raises(RuntimeError) as context:
+            parallel.tick_once()
+            print("RuntimeError has message with substring 'SuccessOnSelected'")
+            assert("SuccessOnSelected" in str(context.exception))
+
+
+def test_parallel_synchronisation():
+    console.banner("Parallel Synchronisation")
+    root = py_trees.composites.Parallel(
+        name="Parallel",
+        policy=py_trees.common.ParallelPolicy.SuccessOnAll(
+            synchronise=True
+        ))
+    success = py_trees.behaviours.Success()
+    success_every_second = py_trees.decorators.FailureIsRunning(
+        name="SuccessEverySecond",
+        child=py_trees.behaviours.SuccessEveryN(name="Flipper", n=2)
+    )
+
+    root.add_children([success, success_every_second])
+    py_trees.display.print_ascii_tree(root)
+    visitor = py_trees.visitors.DebugVisitor()
+    py_trees.tests.tick_tree(root, 1, 1, visitor, print_snapshot=True)
+
+    print("\n--------- Assertions ---------\n")
+    print("root.status == py_trees.common.Status.RUNNING")
+    assert(root.status == py_trees.common.Status.RUNNING)
+    print("success.status == py_trees.common.Status.SUCCESS")
+    assert(success.status == py_trees.common.Status.SUCCESS)
+    print("success_every_second.status == py_trees.common.Status.RUNNING")
+    assert(success_every_second.status == py_trees.common.Status.RUNNING)
+
+    py_trees.tests.tick_tree(root, 2, 2, visitor, print_snapshot=True)
+
+    print("\n--------- Assertions ---------\n")
+    print("root.status == py_trees.common.Status.SUCCESS")
+    assert(root.status == py_trees.common.Status.SUCCESS)
+    print("success.status == py_trees.common.Status.SUCCESS")
+    assert(success.status == py_trees.common.Status.SUCCESS)
+    print("success_every_second.status == py_trees.common.Status.SUCCESS")
+    assert(success_every_second.status == py_trees.common.Status.SUCCESS)
+    # check that succeess did not get ticked
```

Comment:
We are not checking this here, needs to be checked. Perhaps we use a Debug Counter type that is subclassed from behaviours.Success and does a count whenever update() is called? The test should verify that update was called only once. It should reset on initialise()

Reply:
Ah, forgot this...thanks for catching it.

Reply:
Snapshot visitor is the one that can do it, but I need to rejig the test machinery to accept two visitors.

Reply:
For this PR can we just use a derived class? Maybe file an issue and change that in it's own follow up PR?

Reply:
Note, the count method is a bit volatile as it relies on having the rest of the tree staying the same 
 which is usually independent of the behaviour under test. Testing oneshot decorator and idiom are good examples of this - the test should just focus on the properties of oneshots.